extension-workspace-reference.md

```diff
@@ -0,0 +1,82 @@
+# workspace/reference extension to LSP
+
+The `workspace/reference` extension to the Language Server Protocol (LSP) enables a language server to export all of the references to dependencies that code inside of the workspace makes.
+
+Use case: clients of a language server can invoke `workspace/reference` in order to find external references to dependencies. This information can then be stored inside of a database, which allows the caller to create a 'global mapping' of symbols in dependencies to the workspace they are used in (e.g. to see "how do other people use this symbol?").
+
+### Initialization
+
+`ServerCapabilities` may contain a new field to indicate server-side support for this extension:
+
+```typescript
+interface ServerCapabilities {
+  /* ... all fields from the base ServerCapabilities ... */
+
+  /**
+   * The server provides workspace reference exporting support.
+   */
+  workspaceReferenceProvider?: boolean;
```

Comment:
IMO our spec here is extremely confusing from a naming convention standpoint...

- In official LSP, `workspace/symbol` uses the name `workspaceSymbolProvider`
- In official LSP, `textDocument/definition` uses the name `definitionProvider`
- Our `workspace/xfiles`uses `xfilesProvider` not `xworkspaceFilesProvider` or `workspaceXFilesProvider`

I guess given the last point above, I should name this `xreferenceProvider` despite the method name being `workspace/xreference`? This is slightly concerning to me because `workspace/xreference` reads like "workspace cross reference" not "workspace extension reference"..

We could go with `xworkspaceReferenceProvider` and `xworkspace/reference` -- Thoughts?